tsdb/load_shards.go

```diff
@@ -0,0 +1,304 @@
+package tsdb
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/influxdata/influxdb/logger"
+	"github.com/influxdata/influxdb/pkg/limiter"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// res holds the result from opening each shard in a goroutine
+type res struct {
+	s   *Shard
+	err error
+}
+
+func (s *Store) loadShards() error {
+	// Limit the number of concurrent TSM files to be opened to the number of cores.
+	s.EngineOptions.OpenLimiter = limiter.NewFixed(runtime.GOMAXPROCS(0))
```

Comment:
I'm not a fan of pulling `loadShards` and friends into a new file. Pulling methods out of the main implementation file makes it difficult to navigate. It also made it difficult for me to line up diffs, which could also mean issues cherry-picking in the future.